app/src/test/java/com/x8bit/bitwarden/data/platform/processor/BridgeServiceProcessorTest.kt

```diff
@@ -1,52 +1,212 @@
 package com.x8bit.bitwarden.data.platform.processor
 
 import android.os.Build
+import android.os.RemoteCallbackList
+import com.bitwarden.bridge.IBridgeService
+import com.bitwarden.bridge.IBridgeServiceCallback
+import com.bitwarden.bridge.model.EncryptedSharedAccountData
+import com.bitwarden.bridge.model.SharedAccountData
+import com.bitwarden.bridge.util.NATIVE_BRIDGE_SDK_VERSION
+import com.bitwarden.bridge.util.encrypt
+import com.bitwarden.bridge.util.generateSecretKey
+import com.bitwarden.bridge.util.toFingerprint
+import com.bitwarden.bridge.util.toSymmetricEncryptionKeyData
+import com.x8bit.bitwarden.data.platform.base.FakeDispatcherManager
 import com.x8bit.bitwarden.data.platform.manager.FeatureFlagManager
 import com.x8bit.bitwarden.data.platform.manager.model.FlagKey
+import com.x8bit.bitwarden.data.platform.repository.BridgeRepository
+import com.x8bit.bitwarden.data.platform.util.asSuccess
 import com.x8bit.bitwarden.data.platform.util.isBuildVersionBelow
+import io.mockk.coEvery
+import io.mockk.coVerify
 import io.mockk.every
 import io.mockk.mockk
+import io.mockk.mockkConstructor
 import io.mockk.mockkStatic
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
 import org.junit.jupiter.api.Assertions.assertNotNull
 import org.junit.jupiter.api.Assertions.assertNull
 import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
 import org.junit.jupiter.api.Test
 
 class BridgeServiceProcessorTest {
 
-    private val featureFlagManager: FeatureFlagManager = mockk()
+    private val featureFlagManager = mockk<FeatureFlagManager>()
+    private val bridgeRepository = mockk<BridgeRepository>()
 
-    private val bridgeServiceManager = BridgeServiceProcessorImpl(
-        featureFlagManager = featureFlagManager,
-    )
+    private lateinit var bridgeServiceProcessor: BridgeServiceProcessorImpl
 
     @BeforeEach
     fun setup() {
-        mockkStatic(::isBuildVersionBelow)
+        bridgeServiceProcessor = BridgeServiceProcessorImpl(
+            bridgeRepository = bridgeRepository,
+            featureFlagManager = featureFlagManager,
+            dispatcherManager = FakeDispatcherManager(),
+        )
     }
 
     @Test
     fun `when AuthenticatorSync feature flag is off, should return null binder`() {
+        mockkStatic(::isBuildVersionBelow)
```

Comment:
This should be unmocked in the tearDown

Reply:
Aye aye!

e9863c266fb2e2f6e132bad665ae34182c8e9649